modules/ROOT/pages/batch-filters-and-batch-aggregator.adoc

```diff
@@ -250,7 +244,7 @@ For example, if you need to write millions of records to a CSV file, you can pro
 </batch:job>
 ----
 
-Remember that since this batch aggregator is streaming, you can only access its content sequentially:
+You can modify the records in the stream. However, only sequential access to records is possible when streaming within the aggregator. Random access is not supported when streaming because record payloads for random access are exposed as an `ImmutableList`. Streaming through an aggregator requires access to the entire set of records. Without a fixed size. Mule cannot guarantee that all records will fit in memory.
 
 [source,xml,linenums]
 ----
```

Comment:
We also need to test and clarify how this example works, and we shouldn't use "foo":

[source,xml,linenums]
----
<batch:job jobName="batchJob">
	<batch:process-records>
		<batch:step name="batchStep">
			<batch:aggregator doc:name="batchAggregator" size="10">
				<foreach doc:name="For Each">
					<script:execute engine="groovy">
			    	<script:code>
			        		vars['marco'] = 'polo'
					        vars['record'].payload = 'hello'
			    	</script:code>
					</script:execute>
				</foreach>
			</batch:aggregator>
		</batch:step>
	</batch:process-records>
</batch:job>
----